### PR TITLE
Batch performance

### PR DESCRIPTION
**Pull request by BaumiCoder**
Improving performance of PREDICT of many files and folders:

1. Use batch encoding
2. Move the loop for the files into the modul_runner (only one initialization of the model necessary)
3. Support more than one file / folder as input to increase the batch size (also adding file / folder names to output, to make it easier to match the outputs with the list of input files and folders)

Already step 1 decreased the execution time in my situation (almost 100 folders). With CPU execution the decrease if about 50 % and with GPU execution it is about 25 %. (For the other steps I am actually adjusting my shell scripts to use these improvements.)

I adjusted the tests to fit the new parameter types. However, they're still failing (score differs slightly), but they had already failed before my changes. @LuKrO2011 Were there some breaking changes (e.g., new model files)? When this is fixed: Should I add tests for predicting several input folders?

### CODE REVIEW

**Comment by LuKrO2011** [comment]
I appreciate all the changes!

Regarding the tests: The problem seems to be regarding precision of float values. I think it is sufficient to do something like `assert score > 0.99`. If you are fine with that, feel free to add this to the changes, or should I do it directly on the main branch?

>  Should I add tests for predicting several input folders?

A single test doing this in `test_main.py` would be great. Feel free to create necessary files/folders in the `tests/res`. Try to keep it minimal: Is one folder with two subfolders with a single file each sufficient?

**Comment by BaumiCoder** [comment]
I adjusted the `assert` for both old tests. They now only check for the first two digits after the decimal point. 

The same approach uses the new test. New files were not necessary. I used some of the existing files and folders.

**Comment by BaumiCoder** [comment]
As a last performance improvement I replaced the loop for predicting itself with a batch processing.

After all changes the execution time in my situation decreased by about 75 %. The different between CPU and GPU execution is almost 0. The CPU is a bit faster, but I think the GPU only the needs a bigger dataset than 100 folders, to get going.

File: src/readability_classifier/encoders/dataset_encoder.py [hunk 1]
```diff
@@ -56,9 +56,11 @@ def encode_dataset(self, unencoded_dataset: list[dict]) -> ReadabilityDataset:
         bert_dataset = self.bert_encoder.encode_dataset(unencoded_dataset)
         image_dataset = self.visual_encoder.encode_dataset(unencoded_dataset)
 
-        # Normalize the scores
-        scores = [sample["score"] for sample in unencoded_dataset]
-        encoded_scores = self._encode_scores_class(scores)
+        # Normalize the scores if they exist
+        encoded_scores = ["" for _ in range(len(matrix_dataset))]
+        if "scores" in unencoded_dataset[0]:
```
Comment (LuKrO2011) [inline src/readability_classifier/encoders/dataset_encoder.py:61]: Singular:
if "score" in unencoded_dataset[0]:

**Review by LuKrO2011 — CHANGES_REQUESTED** [review]
Bug that makes training crash, easy to resolve.

File: src/readability_classifier/encoders/dataset_encoder.py (hunk 1 above)
Comment (BaumiCoder) [reply src/readability_classifier/encoders/dataset_encoder.py]: @LuKrO2011 Fixed with 7cc9b639b9f431d53429ce78914574a9591e3ef2

**Pull request merged**